# Name : Samarth Deshpande
# Student_id :  8894058

# Importing torch

In [9]:
import torch

# Calculate the dot product between two vectors each of length 3 using pytorch

In [4]:
a = torch.tensor([21, 22, 23])
b = torch.tensor([25, 26, 27])
dot_product = torch.dot(a, b)
print(dot_product)

tensor(1718)


The code uses torch.tensor to create two sets of numbers, a and b, representing vectors.
torch.dot computes a special multiplication between these vectors called the dot product.
The result, shown as tensor(1718), is just  number representing the outcome of this multiplication.

# Calculate the matrix multiplication of two 2-dimensional matrices, each with a size of (2,2) using pytorch

In [5]:
matrix_a = torch.tensor([[5, 6], [7, 8]])
matrix_b = torch.tensor([[9, 10], [8, 16]])
result_matrix = torch.matmul(matrix_a, matrix_b)
print(result_matrix)

tensor([[ 93, 146],
        [127, 198]])


torch.matmul is used to perform matrix multiplication bterween matrix_a and matrix_b
The final output is a Pytorch tensor representing matrix product

# Calculate the dot product between the same vectors as above (re-define them in Tensorflow) 

In [10]:
import tensorflow as tf

In [11]:
# Redefining vectors for dot product
vector1 = tf.constant([21, 22, 23], dtype=tf.int32)
vector2 = tf.constant([25, 26, 27], dtype=tf.int32)

In [12]:
# Calculating dot product
dot_product = tf.tensordot(vector1, vector2, axes=1)
print("Dot Product:", dot_product.numpy())

Dot Product: 1718


To compute the dot product between two vectors we use the tf.tensordot 

# Calculate the matrix multiplication of the same matrics as above (re-define them in Tensorflow) 

In [16]:
# Redefining matrices for matrix multiplication
matrix_1 = tf.constant([[5, 6], [7, 8]], dtype=tf.int32)
matrix_2 = tf.constant([[9, 10], [8, 16]], dtype=tf.int32)
# Calculating matrix multiplication
matrix_product = tf.matmul(matrix_1, matrix_2)
print(matrix_product.numpy())

[[ 93 146]
 [127 198]]


tf.matmul function is used to perform matrix multiplication.